testutil/network/genesis.go

```diff
@@ -0,0 +1,30 @@
+package network
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/cosmos/gogoproto/proto"
+	"github.com/stretchr/testify/require"
+)
+
+// GetGenesisState retrieves the genesis state for a given module from the underlying cosmos-sdk in-memory network.
+func GetGenesisState[T proto.Message](t *testing.T, moduleName string, memnet InMemoryNetwork) T {
+	t.Helper()
+
+	// Ensure the in-memory network has been started.
+	_ = memnet.GetNetwork(t)
+
+	var genesisState T
+	// NB: As this function is generic, it MUST use reflect in order to unmarshal
+	// the genesis state as the codec requries a reference to a concrete type pointer.
+	genesisStateType := reflect.TypeOf(genesisState)
```

Comment:
Not sure of this but can we not use the gateway types codec to deserialise this without reflect. Cosmos has an interface registry for eacd module I don't know if this includes the genesis state.

Reply:
I'm not sure what exactly you're referring to but if it requires passing an allocated but unassigned pointer, it will have the same issue. Link?

Reply:
Nvm the codec doesn't cover the GenesisState object my bad.